### DISCUSSION TITLE
Vector supports multiple config files

### DISCUSSION THREAD

**Discussion by Joshualy94** [vectordotdev/vector#20123] [Q&A] [answered]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Use Cases

Hi, I wonder if I understand it correctly, but I am using vector with all components in one config file (vector.yaml at /etc/vector), which makes it too large to handle a large system. Is there a way we can store the structure in several config files and make the structure more clear?

Thanks!

**Comment by andjhop** [comment]
Yes: https://vector.dev/docs/reference/configuration/#multiple-files.

**Reply by Joshualy94** [reply]
Thank you for the reply. I have seen this before. However, I start vector service by running `sudo systemctl start vector`, which will use vector.yaml at /etc/vector by default. 

I didn't find a way that I can run several vector config files in this way.
Appreciate it there's any suggestion or best practice on it. Since I am managing a large system, there are a lot of components. Is it better to separate them into several config files?

**Reply by polarathene** [reply]
> I didn't find a way that I can run several vector config files in this way.

`systemctl edit vector` should let you add your own override if you don't have direct control over the systemd unit config (_like when it's provided via a package_). That should persist across updates and allow you to override the command run so that you can follow the advice in the linked docs.

**Comment by tbenade** [comment, marked as answer]
Hi @Joshualy94 ,
Multi file configuration is well supported and we successfully deploy the [namespaced multifile functionality ](https://vector.dev/highlights/2021-11-18-implicit-namespacing/) using [VECTOR_CONFIG_DIR](https://vector.dev/docs/reference/cli/#vector_config_dir) setting.

You don't need follow the namespaced practice you can simply use multiple files. https://vector.dev/docs/reference/configuration/#multiple-files.

You haven't mentioned how you are running/starting vector. If you are simply running it as an application you can invoke vector with `--config-dir <your-config-directory>`

If using systemd and unit files you will need set the environment variable `VECTOR_CONFIG_DIR` in the vector process by supplying a unit `EnvironmentFile` prior to starting the process.

`# file: vector.default`
```
VECTOR_CONFIG_DIR=/etc/vector
VECTOR_LOG=INFO
VECTOR_COLOR=never
```

Then put the above here `mv <whatever you call the above file> /etc/default/vector`.

Hope that helps.
